libs/isograph-react/src/core/garbageCollection.ts

```diff
@@ -42,7 +45,7 @@ export function retainQuery(
 }
 
 export function garbageCollectEnvironment(environment: IsographEnvironment) {
-  const retainedIds = new Set<DataId>([ROOT_ID]);
+  const retainedIds: RetainedIds = { Query: new Set<DataId>([ROOT_ID]) };
```

Comment:
This is not good, because `Query` is hard coded, can we garbage collect it?

Reply:
Happy to revisit this in the future. I'm not concerned right now. I don't think we would ever garbage collect `Query` or `Query[ROOT_ID]`, because (in the current version) we keep 10 queries in the GC buffer.

That being said, that all needs to be more customizable, but we can revisit it in the future.

Another thing we could do is to treat `Query[ROOT_ID]` as "special", and treat each field on `Query[ROOT_ID]` as its own garbage-collectable object. Because, right now, since everything goes through `Query[ROOT_ID]` we will eventually have a `Query[ROOT_ID]` object with the root fields of every single query you ever made. Oh well!

Reply:
Right now it will `garbageCollect` if different type than `Query` is used

Reply:
Oh do we have a bug with the star wars API example in the quickstart then?

Reply:
I've fixed it

Reply:
given the change, can we make `const retainedIds: RetainedIds = {}`?

Reply:
Yes, but this required a test change

Reply:
nice lgtm